imap_processing/idex/idex_l1b.py

```diff
@@ -324,3 +337,62 @@ def compute_trigger_values(
             f"there is only one valid trigger value per event. This "
             f"caused Merge Error: {e}"
         ) from e
+
+
+def get_spice_data(
+    epoch: np.ndarray, idex_attrs: ImapCdfAttributes
+) -> dict[str, xr.DataArray]:
+    """
+    Use spice to query ephemeris, attitude, celestial coordinates for each dust event.
+
+    Parameters
+    ----------
+    epoch : np.ndarray
+        Dust impact times.
+    idex_attrs : ImapCdfAttributes
+        CDF attribute manager object.
+
+    Returns
+    -------
+    dict
+        Spice array names and xr.DataArrays.
+    """
+    # 'epoch' is in nanoseconds since j2000
+    # convert epoch times to seconds since j2000
+    et = j2000ns_to_j2000s(epoch)
+    # Get IDEX rotation matrix (matrix to get transformation from the 'ECLIPJ2000'
+    # Reference frame to the IDEX instrument frame)
+    rotation = get_rotation_matrix(et, SpiceFrame.ECLIPJ2000, SpiceFrame.IMAP_IDEX)
+    # In order yaw, pitch, roll in radians
+    euler_angles = rotation_matrix_to_euler_angles(rotation)
+
+    # Get position and velocity of IMAP
+    ephemeris = imap_state(et, observer=SpiceBody.SUN)
+    imap_position = ephemeris[:3]
+
+    # get right ascension and declination of the IMAP spacecraft
```

Comment:
Little nitpicky - this comment seems unnecessary to me since the following function is named so specifically.